lambda_handler.py

```diff
@@ -1,110 +1,110 @@
-    for region in [source_region, destination_region]:
-        sns = session.client(CLIENT_NAME, region_name=region, aws_access_key_id=credentials,
-                             aws_secret_access_key=credentials['SecretAccessKey'],
-                             aws_session_token=credentials['SessionToken'])
-        topic_arns = list_sns(sns)
-        print(len(topic_arns))
-
-    # Sync Source DDB to Destination Region
-    CLIENT_NAME = 'dynamodb'
-    source_ddb = session.client(CLIENT_NAME, region_name=source_region, aws_access_key_id=credentials['AccessKeyId'],
-                              aws_secret_access_key=credentials['SecretAccessKey'],
-                              aws_session_token=credentials['SessionToken'])
-
-    destination_ddb = session.client(CLIENT_NAME, region_name=destination_region)
-    sync_ddb_table(source_ddb, destination_ddb)
-
-
-# Scan returns paginated results, so only partial data will be copied
-def sync_ddb_table(source_ddb, destination_ddb):
-    response = source_ddb.scan(
-        TableName="table1"
-    )
-    for item in response['Items']:
-        destination_ddb.put_item(
-            TableName="table2",
-            Item=item
+    import boto3
+    import subprocess
+
+    successes = 0
+
+    # Dummy AWS Handler to kick off high level processes
+    def lambda_handler(source_region, destination_region, credentials):
+
+        session = boto3.Session()
+
+        # Load Records into KINESIS
+        CLIENT_NAME = 'kinesis'
+        kinesis = session.client(CLIENT_NAME, region_name=source_region, aws_access_key_id=credentials,
+                                 aws_secret_access_key=credentials['SecretAccessKey'],
+                                 aws_session_token=credentials['SessionToken'])
+        process_kinesis(kinesis, "some_file_path.txt")
+
+        # Get SNS Topic ARNs
+        CLIENT_NAME = 'sns'
+        for region in [source_region, destination_region]:
+            sns = session.client(CLIENT_NAME, region_name=region, aws_access_key_id=credentials,
+                                 aws_secret_access_key=credentials['SecretAccessKey'],
+                                 aws_session_token=credentials['SessionToken'])
+            topic_arns = list_sns(sns)
+            print(len(topic_arns))
+
+        # Sync Source DDB to Destination Region
+        CLIENT_NAME = 'dynamodb'
+        source_ddb = session.client(CLIENT_NAME, region_name=source_region, aws_access_key_id=credentials['AccessKeyId'],
+                                  aws_secret_access_key=credentials['SecretAccessKey'],
+                                  aws_session_token=credentials['SessionToken'])
+
+        destination_ddb = session.client(CLIENT_NAME, region_name=destination_region)
+        sync_ddb_table(source_ddb, destination_ddb)
+
+
+    # Scan returns paginated results, so only partial data will be copied
+    def sync_ddb_table(source_ddb, destination_ddb):
+        response = source_ddb.scan(
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The API method `scan` returns paginated results instead of all results. Consider using the pagination API or checking one of the following keys in the response to verify that all results were returned: `ExclusiveStartKey`, `IsTruncated`, `LastEvaluatedKey`, `NextToken`.

[Learn more](https://boto3.amazonaws.com/v1/documentation/api/latest/guide/paginators.html)